### ISSUE TITLE
Cannot call model.train more than once

### ISSUE THREAD

**Issue by jbk41** [matterport/Mask_RCNN#784]
Hi, I am trying to use a bootleg adam optimizer by just calling different runs to model.train using decreasing learning rates in each call to model.train. (I used epochs=1 for testing purposes)

``` 
        t1s = time.time()
        model.train(dataset_train, dataset_test, 
                    learning_rate=config.LEARNING_RATE / 10, 
                    augmentation=augmentation,
                    epochs=1, 
                    layers="all")
        t1e = time.time()
        print(t1e-t1s)

        t2s = time.time()
        model.train(dataset_train, dataset_test, 
                    learning_rate=config.LEARNING_RATE / 100,
                    augmentation=augmentation,
                    epochs=1, 
                    layers="all")
        t2e = time.time()
        print(t2e-t2s)

        t3s = time.time()
        model.train(dataset_train, dataset_test, 
                    learning_rate=config.LEARNING_RATE / 1000,
                    augmentation=augmentation,
                    epochs=1, 
                    layers="all")
        t3e = time.time()
        print(t3e-t3s)    
```
After this call, the checkpoint log prints `Checkpoint Path: /home/logs/cells20180717T1057/mask_rcnn_cells_{epoch:04d}.h5`
and continues to train and logs the model with these lines: 
```
Epoch 1/1
- 218s - loss: 5.0277 - rpn_class_loss: 0.2963 - rpn_bbox_loss: 1.7549 - mrcnn_class_loss: 0.3324 - mrcnn_bbox_loss: 0.8074 - mrcnn_mask_loss: 1.8368 - val_loss: 4.1267 - val_rpn_class_loss: 0.2864 - val_rpn_bbox_loss: 1.2851 - val_mrcnn_class_loss: 0.1385 - val_mrcnn_bbox_loss: 0.6875 - val_mrcnn_mask_loss: 1.7292

Epoch 00001: val_loss improved from inf to 4.12666, saving model to /home/logs/cells20180717T1057/mask_rcnn_cells_0001.h5
```
This is all happening within the first code block listed above and the time print statement says one epoch took about 300 seconds.

The problem arises during the next model.train call. It logs the same `Checkpoint Path: /home/logs/cells20180717T1057/mask_rcnn_cells_{epoch:04d}.h5` line, but then it does not print any statement about the epoch number or loss values. Instead it continues immediately to the third call to model.train which also does not show any signs of training. Both the second and third calls to model.train only took about 30 seconds, which is significantly shorter than the time it took to run the first call so I suspect that the model is not training at all during the second and third calls to model.train.

My script does not crash and I haven't changed the `set_log_dir` function in model.py. I have only modified verbosity modes. I know that I can just add an adam optimizer to the keras model but I would also like to run additional calls to model.train but using layers='heads' instead of 'all'

Has anyone else had this issue? Thanks.

**Comment by Paulito-7** [comment]
You wrote `epochs=1` for each stage of the training whereas you should cumulate values when going to further steps
E.g for second call to `train` you should have `epochs=2`

**Comment by coreywho** [comment]
Alternatively, you can add this line into model.py

**Comment by HAMZARaouia** [comment]
You can change the number of the epochs. For example, `epochs = 3` for 3 runs on the training set.
